File: src/main/java/raylras/zen/code/type/AnyType.java
package raylras.zen.code.type;

import raylras.zen.code.CompilationEnvironment;

public class AnyType extends Type {

    public static final AnyType INSTANCE = new AnyType();

    @Override
    public boolean isCastableTo(Type type, CompilationEnvironment env) {
        return true;
    }

    @Override
    public String toString() {
        return "any";
    }

}

File: core/src/main/kotlin/raylras/intellizen/type/AnyType.kt
package raylras.intellizen.type

import raylras.intellizen.CompilationEnvironment

object AnyType : Type {
    override val typeName = "any"

    override fun isSupertypeTo(type: Type) = true

    override fun isCastableTo(that: Type?, env: CompilationEnvironment) = true

    override fun toString() = typeName
}
